streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -795,6 +811,29 @@ public Sensor addLatencySensor(String scopeName, String entityName, String opera
             return sensor;
         }
 
+        @Override
+        public Sensor addCountSensor(String entityName, String operationName, String... tags) {
+            // extract the additional tags if there are any
+            Map<String, String> tagMap = new HashMap<>(this.metricTags);
+            if ((tags.length % 2) != 0)
+                throw new IllegalArgumentException("Tags needs to be specified in key-value pairs");
+
+            for (int i = 0; i < tags.length; i += 2)
+                tagMap.put(tags[i], tags[i + 1]);
+
+            String metricGroupName = "stream-thread-cache-metrics";
+
+            Sensor sensor = metrics.sensor(sensorNamePrefix + "-" + entityName + "-" + operationName);
+            addCountMetrics(metricGroupName, sensor, entityName, operationName, tagMap);
+            return sensor;
+
+        }
+
+        private void addCountMetrics(String metricGrpName, Sensor sensor, String entityName, String opName, Map<String, String> tags) {
```

Comment:
EDIT: actually since we use sampled stats with a window `Count` will not be ever increasing, but will be increasing for a while, then suddenly reset to a smaller value when a window as elapsed. But still I feel this is not the right metric that we want.